File: scripts/clean.ts
import fs from 'node:fs'

import { getWorkspaceFiles } from './pnpm'

const dirsToRemove = [
  '.output',
  '.data',
  '.nuxt',
  '.nitro',
  '.cache',
  'dist',
  'node_modules',
  'playwright-report',
  'test-results'
]

const filesToRemove = [
  'package.json'
]

dirsToRemove.forEach((dir) => {
  const allPackageDirs = getWorkspaceFiles(dir)

  allPackageDirs.forEach(pkgDir => fs.rmSync(pkgDir, { recursive: true, force: true }))
})

filesToRemove.forEach((filename) => {
  const allPackageFiles = getWorkspaceFiles(filename)

  allPackageFiles.forEach(file => fs.rmSync(file))
})

File: scripts/clean.rb
require 'fileutils'
require 'optparse'

require_relative 'utils/workspace'

options = {}
OptionParser.new do |parser|
  parser.on("--lockfile", "Remove lockfile in addition to other files")
end.parse(ARGV, into: options)

entries_to_remove = [
  # Folders
  '.output',
  '.data',
  '.nuxt',
  '.nitro',
  '.cache',
  'dist',
  'playwright-report',
  'test-results',

  # Files
  'package.json'
]

system "pnpm pm clean #{"--lockfile" if options[:lockfile]}"

Workspace.packages.each do |package|
  entries_to_remove.each do |entry|
    FileUtils.remove_entry_secure(package / entry, force = true)
  end
end
